railties/lib/rails/generators/actions.rb

```diff
@@ -39,9 +39,10 @@ def gem(*args)
 
         in_root do
           str = "gem #{parts.join(", ")}"
-          str = indentation + str
-          str = "\n" + str
-          append_file "Gemfile", str, verbose: false
+          gsub_file "Gemfile", /(.+|\n)\z/, verbose: false do |match|
+            match << "\n" unless match.end_with?("\n")
+            match << indentation + str + "\n"
+          end
```

Comment:
I think this might simplify to:

```ruby
gsub_file "Gemfile", /\n?\z/, verbose: false do
  "\n#{indentation}#{str}\n"
end
```

Or we might even want to consider:

```ruby
gsub_file "Gemfile", /(\n)?\z/, verbose: false do |match|
  "\n#{indentation}#{str}#{match}"
end
```

Either way, it's probably worth extracting a helper -- because that way yeah we'll be able to fix up the `end`s to match.

I think some of the other tests' expectations need the extra trailing `\n` too, to show that this is doing what we want (e.g. `test_gem_should_put_gem_dependency_in_gemfile`)